File: src/test/java/net/sf/rails/game/state/CountableItemImpl.java
package net.sf.rails.game.state;

import net.sf.rails.game.state.Countable;
import net.sf.rails.game.state.CountableItem;
import net.sf.rails.game.state.Item;

class CountableItemImpl extends CountableItem<Countable> {
    
    protected CountableItemImpl(Item parent, String id) {
        super(parent, id, Countable.class);
    }

    static CountableItemImpl create(Item parent, String id) {
        return new CountableItemImpl(parent, id);
    }
    
}

File: dotnet/GameStateTest/CountableItemImpl.cs
using GameLib.Net.Game.State;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GameStateTest
{
    public class CountableItemImpl : CountableItem<ICountable>
    {
        protected CountableItemImpl(IItem parent, string id) : base(parent, id)
        {
        }

        public static CountableItemImpl Create(IItem parent, string id)
        {
            return new CountableItemImpl(parent, id);
        }
    }
}
